kolibri/core/content/serializers.py

```diff
@@ -248,3 +252,60 @@ def get_updated_resource(self, instance):
 
     def get_is_leaf(self, instance):
         return instance.kind != content_kinds.TOPIC
+
+
+class ContentDownloadRequestMetadataSerializer(serializers.Serializer):
+
+    title = serializers.CharField()
+    file_size = serializers.IntegerField()
+    learning_activities = serializers.ListField(
+        child=serializers.ChoiceField(learning_activities.choices)
+    )
+
+
+class ContentDownloadRequestSeralizer(serializers.ModelSerializer):
+
+    metadata = ContentDownloadRequestMetadataSerializer()
+
+    class Meta:
+
+        model = ContentDownloadRequest
+        fields = (
+            "id",
+            "contentnode_id",
+            "metadata",
+        )
+
+    def create(self, validated_data):
+        # if there is an existing deletion request, delete the deletion request
+        if "request" in self.context and self.context["request"].user is not None:
+            user = self.context["request"].user
+        else:
+            raise serializers.ValidationError("User must be defined")
+
+        deletion_request = ContentRemovalRequest.objects.filter(
+            contentnode_id=validated_data["contentnode_id"],
+            source_id=user.id,
+        )
```

Comment:
Add filter `reason=ContentRequestReason.UserInitiated`. Without that, we would lose tracking that the `SyncInitiated` a removal when the resource was no longer required. Even if a `SyncInitiated` removal exists, the new `UserInitiated` download that will be added here will prevent that removal from being processed.

Ideally, we also have `source_model=FacilityUser.morango_model_name` here too. It's unlikely we would get collisions for a `source_id` but it makes sense.

It's probably worth making a helper method on the model manger / queryset to make this filtering easier and consolidated, similar to `build_for_user`, but that's out of scope. I can implement that later and refactor this.